### PR TITLE
initial changes for cluster placement support

### PR DESCRIPTION
**Pull request by gparvin**
Signed-off-by: Gus Parvin <gparvin@redhat.com>

Placement support user story: https://github.com/open-cluster-management/backlog/issues/14907

Note that community support is a different user story so no testing has been done without App lifecycle CRDs.  More changes could needed for that.

### CODE REVIEW

**Review by JustinKuli — COMMENTED** [review]
Is it usual to copy the CRDs from other repos into ours? For example, the cluster placement CRD. Just wondering why we can't add that repo as a dependency, and then we don't have to worry about falling out of sync?

Looks fine to me, but big enough that I'd like another reviewer to approve :)

File: pkg/controller/propagator/propagation.go [hunk 1]
```diff
@@ -210,6 +201,89 @@ func (r *ReconcilePolicy) handleRootPolicy(instance *policiesv1.Policy) error {
...
+		PlacementBinding: pb.GetName(),
+		PlacementRule:    plr.GetName(),
+		// Decisions:        plr.Status.Decisions,
+	}
+	return plr.Status.Decisions, placement, nil
+}
+
+// getClusterPlacementDecisions return the placement decisions from cluster
+// placement decisions
+func getClusterPlacementDecisions(c client.Client, pb policiesv1.PlacementBinding, instance *policiesv1.Policy) ([]appsv1.PlacementDecision, *policiesv1.Placement, error) {
```
[the remark this answers is not in the document]
Comment (gparvin) [reply pkg/controller/propagator/propagation.go:226]: Yes, this is what I was referring to when I mentioned community placement work for sprint 2 may cause some more changes.  I didn't have a good idea how to improve it now so I'll see if that work drives a change.

File: pkg/controller/propagator/propagation.go [hunk 2]
```diff
@@ -210,6 +201,89 @@ func (r *ReconcilePolicy) handleRootPolicy(instance *policiesv1.Policy) error {
...
+		return d, placement, nil
+	} else if pb.PlacementRef.APIGroup == clusterv1alpha1.SchemeGroupVersion.Group &&
+		pb.PlacementRef.Kind == clusterv1alpha1.Kind {
+		d, placement, err := getClusterPlacementDecisions(c, pb, instance)
+		if err != nil {
+			return nil, nil, err
+		}
+		return d, placement, nil
+	}
+	return nil, nil, nil
```
[the remark this answers is not in the document]
Comment (gparvin) [reply pkg/controller/propagator/propagation.go:284]: You are correct.  I'll make an update here.
/hold

[the remark this answers is not in the document]
Comment (gparvin) [reply pkg/controller/propagator/propagation.go]: Update made

File: pkg/controller/propagator/propagation.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gparvin) [reply pkg/controller/propagator/propagation.go:226]: I did notice I had 2 copies of the crds.  I got rid of one of them.

**Comment by gparvin** [comment]
There were no new changes since the lgtm so merging.

**Pull request merged**